spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/TaskControllerTests.java

```diff
@@ -151,6 +157,59 @@ public void testSave() throws Exception {
 	}
 
 	@Test
+	public void testCompose() throws Exception {
+		assertEquals(0, repository.count());
+		TaskDefinition testTask = new TaskDefinition("task", "task");
+		repository.save(testTask);
+		appRegistry.save("task", ApplicationType.task, new URI("http://fake.example.com/"),null);
+		mockMvc.perform(
+				post("/tasks/definitions/compose").param("name", "myTask").param("definition", "task")
+						.accept(MediaType.APPLICATION_JSON)).andDo(print())
+				.andExpect(status().isOk());
+
+		assertEquals(2, repository.count());
+
+		TaskDefinition myTask = repository.findOne("myTask");
+
+		assertEquals(2, myTask.getProperties().size());
+		assertEquals("myTask", myTask.getProperties().get("spring.cloud.task.name"));
+		assertEquals("task", myTask.getProperties().get("graph"));
+		assertEquals("composed-task-runner --graph=\"task\"", myTask.getDslText());
```

Comment:
This would be a good place to test proper escaping, hence calling for a non-trivial composition again